pkg/plugins/v1/scaffolds/internal/templates/dockerfile.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package templates
+
+import (
+	"sigs.k8s.io/kubebuilder/pkg/model/file"
+)
+
+var _ file.Template = &Dockerfile{}
+
+// Dockerfile scaffolds a Dockerfile for building a main
+type Dockerfile struct {
+	file.TemplateMixin
+
+	// HelmOperatorVersion is the version of the base image and operator binary used in the project
+	HelmOperatorVersion string
+}
+
+// SetTemplateDefaults implements input.Template
+func (f *Dockerfile) SetTemplateDefaults() error {
+	if f.Path == "" {
+		f.Path = "Dockerfile"
+	}
+
+	f.TemplateBody = dockerfileTemplate
+
+	return nil
+}
+
+const dockerfileTemplate = `# Build the manager binary
+FROM quay.io/anrastog/helm-operator:{{.HelmOperatorVersion}}
```

Comment:
Since I was not aware of any active tags for the image, I added my own image "quay.io/anrastog/helm-operator:canary" to the generated Dockerfile and have been using it for testing.

I'll test the code with quay.io/joelanford/helm-operator:master, and get back to you on what'sbroken.

Reply:
The `HelmOperatorVersion` is set by `VERSION = $(shell git describe --dirty --tags --always)` in the Makefile right now. We could leave as it is for now.

Reply:
Update: The `quay.io/joelanford/helm-operator` base image only has `:master` tag, so I have hardcoded HelmOperatorVersion to it as suggested.